src/sinks/aws_kinesis/streams/config.rs

```diff
@@ -77,12 +76,6 @@ pub struct KinesisStreamsSinkConfig {
     #[serde(flatten)]
     pub base: KinesisSinkBaseConfig,
 
-    /// The log field used as the Kinesis record’s partition key value.
-    ///
-    /// If not specified, a unique partition key is generated for each Kinesis record.
-    #[configurable(metadata(docs::examples = "user_id"))]
-    pub partition_key_field: Option<ConfigValuePath>,
-
```

Comment:
is it necessary? unfortunately I'm not familiar with rust and my experience in it is only this PR and quick tour, I tried to do it, but don't understand because of lack of experience in rust

Reply:
IMO it is necessary to keep good code hygiene, and this base struct serves no purpose now except to confuse people.

I can take a pass at making this change.

Reply:
if it's okay for you it would be great, thank you! I'm sorry, but I'll spend too much time and delay this changes for a weeks and pretty sure doesn't do it in a correct way

Reply:
thanks for info